### PR TITLE
Feature/slack channel/proxy

### LINKED ISSUE #6123: Slack proxy option support
**Description of Problem**:
The slack channel does not currently support the slack client [proxy](https://github.com/slackapi/python-slackclient#proxy) option. This was discussed on the [forum](https://forum.rasa.com/t/slack-connector-has-to-use-proxy-for-outgoing-requests-in-docker/30845).

**Overview of the Solution**:
Add the client library [proxy](https://github.com/slackapi/python-slackclient#proxy) option to the slack channel [here](https://github.com/RasaHQ/rasa/blob/4830308fcc21ba3c82f296e510ba7544fb5166d4/rasa/core/channels/slack.py#L26) and expose in the `credentials.yml`.

**Definition of Done**:
- [ ] Feature described the docs
- [ ] Feature mentioned in the changlog

### PR DESCRIPTION
**Pull request by iamtofr**
Fixes #6123  
**Proposed changes**:
- adds proxy property to credentials.yml
- make proxy optional
- update docs and changelog

**Status (please check what you already did)**:
- [/] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by sara-tagger** [comment]
Thanks for submitting a pull request 🚀 @amn41 will take a look at it as soon as possible ✨

**Comment by iamtofr** [comment]
#6123

File: changelog/6123.improvement.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+adds support for proxy use in slack channel
```
Comment (ricwo) [inline changelog/6123.improvement.rst:1]:
```suggestion
Add support for proxy use in :ref:`slack` input channel.
```

File: docs/user-guide/connectors/slack.rst [hunk 2]
```diff
@@ -66,11 +66,14 @@ You need to supply a ``credentials.yml`` with the following content:
   "OAuth & Permissions" tab as your ``slack_token``. It should start
   with ``xoxb``.
 
+- Optional you can add a ``proxy`` to route outgoing traffic to the slack API through. Only http proxy is currently supported!
```
Comment (ricwo) [inline docs/user-guide/connectors/slack.rst:69]:
```suggestion
- Optional: You can add a ``proxy`` through which to route outgoing traffic to the Slack API. Only HTTP proxies are currently supported.
```

File: rasa/cli/initial_project/credentials.yml [hunk 3]
```diff
@@ -15,6 +15,7 @@ rest:
 #slack:
 #  slack_token: "<your slack token>"
 #  slack_channel: "<the slack channel>"
+#  proxy: "<your outgoing proxy>"
```
Comment (ricwo) [inline rasa/cli/initial_project/credentials.yml:18]:
```suggestion
#  proxy: "<your HTTP outgoing proxy>"
```

File: rasa/core/channels/slack.py [hunk 4]
```diff
@@ -153,6 +161,7 @@ def __init__(
         """
         self.slack_token = slack_token
         self.slack_channel = slack_channel
+        self.proxy = proxy
```
Comment (ricwo) [inline rasa/core/channels/slack.py:164]: this needs to be added to the `__init__()` docstring

File: docs/user-guide/connectors/slack.rst [hunk 5]
```diff
@@ -66,11 +66,14 @@ You need to supply a ``credentials.yml`` with the following content:
...
+
 .. code-block:: yaml
 
    slack:
      slack_token: "xoxb-286425452756-safjasdf7sl38KLls"
      slack_channel: "C011GR5D33F"
+     proxy: "http://myProxy.online"
```
Comment (ricwo) [inline docs/user-guide/connectors/slack.rst:76]: let's add a comment here as well

**Pull request merged**